File: app/src/main/java/com/tutv/android/ui/series_carrousel/SeriesCarrouselAdapter.java
package com.tutv.android.ui.series_carrousel;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.tutv.android.R;
import com.tutv.android.ui.series_carrousel.tv_banner.TvBannerComponent;
import com.tutv.android.ui.series_carrousel.tv_banner.TvBannerView;

import org.jetbrains.annotations.NotNull;

public class SeriesCarrouselAdapter extends RecyclerView.Adapter {
    private final SeriesCarrouselPresenter presenter;
    private final Context context;

    public SeriesCarrouselAdapter(SeriesCarrouselPresenter presenter, Context context) {
        this.presenter = presenter;
        this.context = context;
    }

    @NonNull
    @NotNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(context);
        View slideLayout = inflater.inflate(R.layout.slider_item, parent, false);

        return new TvBannerComponent(slideLayout);
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull RecyclerView.ViewHolder holder, int position) {
        presenter.onBindRepositoryRowViewAtPosition(position, (TvBannerView) holder);
    }

    @Override
    public int getItemCount() {
        return presenter.getItemCount();
    }
}

File: app/src/main/java/com/tutv/android/ui/series_carrousel/SeriesCarrouselAdapter.kt
package com.tutv.android.ui.series_carrousel

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.tutv.android.R
import com.tutv.android.ui.series_carrousel.tv_banner.TvBannerComponent
import com.tutv.android.ui.series_carrousel.tv_banner.TvBannerView

class SeriesCarrouselAdapter(private val presenter: SeriesCarrouselPresenter?, private val context: Context?) :
        RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(context)
        val slideLayout = inflater.inflate(R.layout.slider_item, parent, false)
        return TvBannerComponent(slideLayout)
    }

    override fun onViewDetachedFromWindow(holder: RecyclerView.ViewHolder) {
        super.onViewDetachedFromWindow(holder)
        presenter?.onViewHolderDetached(holder as TvBannerView)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        presenter?.onBindRepositoryRowViewAtPosition(position, holder as TvBannerView)
    }

    override fun getItemCount(): Int {
        return presenter?.getItemCount() ?: 0
    }

}